modules/account-lib/test/unit/coin/eth2/keyPair.ts

```diff
@@ -0,0 +1,74 @@
+import should from 'should';
+import { KeyPair } from '../../../../src/coin/eth2';
+import * as testData from '../../../resources/eth2/eth2';
+
+const pub = testData.ACCOUNT_1.publicKey;
+const prv = testData.ACCOUNT_1.privateKey;
+
+describe('Eth2 Key Pair', () => {
+  describe('should create a valid KeyPair', () => {
+    it('from an empty value', () => {
+      const keyPair = new KeyPair();
+      should.exists(keyPair.getKeys().privateKey);
+      should.exists(keyPair.getKeys().publicKey);
+      should.equal(
+        keyPair
+          .getKeys()
+          .privateKey.toHexString()
+          .slice(0, 32),
+        testData.BLSPrivKeyPrefix,
+      );
+      should.equal(
+        keyPair
+          .getKeys()
+          .publicKey.toHexString()
+          .slice(0, 24),
+        testData.BLSPubKeyPrefix,
+      );
+    });
+
+    it('from a private key', () => {
+      const keyPair = new KeyPair({ prv: prv });
+      should.equal(keyPair.getKeys().privateKey, prv);
+      should.equal(keyPair.getKeys().publicKey, pub);
+    });
+
+    it('without source', () => {
+      const keyPair = new KeyPair();
+      should.exists(keyPair.getKeys().privateKey);
+      should.exists(keyPair.getKeys().publicKey);
+    });
+
+    it('from a byte array private key', () => {
+      const keyPair = new KeyPair({ prv: Buffer.from(testData.ACCOUNT_1.privateKeyBytes).toString('hex') });
+      should.equal(keyPair.getKeys().privateKey, prv);
+    });
+  });
+
+  describe('should fail to create a KeyPair', () => {
+    it('from a public key', () => {
+      const source = { pub: pub };
+      should.throws(
+        () => new KeyPair(source),
+        e => e.message.includes(testData.errorMessagePublicKeyKeygen),
+      );
+    });
+
+    it('from an invalid private key', () => {
+      const shorterPrv = { prv: '82A34E' };
+      const longerPrv = { prv: prv + '1' };
+      should.throws(
+        () => new KeyPair(shorterPrv),
+        e => e.message === testData.errorMessageInvalidPrivateKey,
+      );
+      should.throws(
+        () => new KeyPair(longerPrv),
+        e => e.message === testData.errorMessageInvalidPrivateKey,
+      );
+      should.throws(
+        () => new KeyPair({ prv: prv + pub }),
+        e => e.message === testData.errorMessageInvalidPrivateKey,
+      );
+    });
+  });
```

Comment:
Add some tests for `recordFromPrivateKey` as well as it's part of the public interface for KeyPairs